<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GeneratorAbstractFactory</class>
 <widget class="QWidget" name="GeneratorAbstractFactory">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>341</width>
    <height>183</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GeneratorAbstractFactory</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="1">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>!!! under construction !!!</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>abstract factory</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QLineEdit" name="lineEditAbstractFactory"/>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>products</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="InterfaceListEditor" name="productEditor">
     <property name="text">
      <string>define products</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_4">
     <property name="text">
      <string>concrete factories</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QPlainTextEdit" name="plainTextEditConcreteFactories"/>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>InterfaceListEditor</class>
   <extends>QPushButton</extends>
   <header>InterfaceListEditor.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
